i#748 license display

srerickson opened the pull request:
Displays the rights as a link to it's corresponding URI.

Fixes #748

cam156 commented on config/travis/solr_wrapper_test.yml:
```diff
@@ -1,9 +1,9 @@
-    persist: false
-    dir: solr/config
-    name: hydra-test
+  persist: false
```
I may be wrong, but aren't these options for the collection?  So shouldn't thee spaces be there?

awead replied:
@cam156 one or more spaces is sufficient. I was just using to for consistency.

srerickson committed:
Use DisplayRightsComponent for licenses

Translates a give license id to its corresponding label, and makes the
link to the label's URI. The component does not render if the id is
missing or invalid.

There was also a bug with license labels. A nil or bogus value would
cause an error. This is now corrected and a nil label is returned.

Co-authored-by: Seth Erickson <sr.erickson@gmail.com>

awead commented on config/authorities/licenses.yml:
```diff
@@ -41,9 +41,12 @@ terms:
     term: CC0 1.0 Universal
     active: true
-  - id: http://www.europeana.eu/portal/rights/rr-r.html
+  - id: https://rightsstatements.org/page/InC/1.0/?language=en
```
@srerickson I'd remove the `/?language=en` at the end of the url.

srerickson replied:
@awead, it's actually necessary: if you remove the URL param, the link won't resolve.

srerickson replied:
actually this works: https://rightsstatements.org/page/InC/1.0/
but not this: https://rightsstatements.org/page/InC/1.0

bad.

awead commented on app/components/display_rights_component.rb:
```diff
@@ -0,0 +1,11 @@
+
+  def initialize(id:)
+    @id = id
+    @label = WorkVersion::Licenses.label(id) if id.present?
```
@srerickson will there even be a nil id? If so, we could use the `render?` method. This is a built-in feature of ApplicationComponent. Basically, if we were to initialize the component without an id, or a `nil` value, it won't display anything:

``` ruby
def render?
  id.present?
end
```

With the way it's written above, if the id were nil, the span html would still render and you'd get some kind of a broken link, I think.

awead commented on app/decorators/resource_decorator.rb:
```diff
@@ -32,6 +32,12 @@ def display_doi
+    return unless respond_to?(:rights)
+
+    DisplayRightsComponent.new(id: rights)
+  end
```
@srerickson I'm curious what errors you were seeing when this was on the WorkVersionDecorator. My guess is that if this is defined on the WorkVersionDecorator, you won't need the return statement, but, I bet you'll need to account for nil ids. That may be why you were seeing errors. See my other comment about the `render?` method. That may solve the problem.

srerickson replied:
@awead here's an example of the error that happens if I move it to WorkVersionDecorator. (This is with the render? method added).

```rb
  1) WorkVersionMetadataComponent rendering renders a string with label
     Failure/Error: value = format_value(work_version.send(attr))
     
     NoMethodError:
       undefined method `display_rights' for #<ResourceDecorator:0x0000562f16f02800>
     # ./app/components/work_version_metadata_component.rb:58:in `block in attributes'
     # ./app/components/work_version_metadata_component.rb:56:in `map'
     # ./app/components/work_version_metadata_component.rb:56:in `attributes'
     # ./app/components/work_version_metadata_component.html.erb:3:in `call'
     # /usr/local/bundle/gems/view_component-2.18.0/lib/view_component/base.rb:78:in `render_in'
     # /usr/local/bundle/gems/view_component-2.18.0/lib/view_component/render_component_helper.rb:6:in `render_component'
     # /usr/local/bundle/gems/view_component-2.18.0/lib/view_component/test_helpers.rb:27:in `render_inline'
     # ./spec/components/work_version_metadata_component_spec.rb:7:in `block (2 levels) in <top (required)>'
     # ./spec/components/work_version_metadata_component_spec.rb:35:in `block (3 levels) in <top (required)>'
```

awead reviewed (changes requested):
@srerickson Looks good! I think you're replicating the right pattern here. Just a few things to tidy up and we'll be able to push this out.
